.gitlab/functional_test/regression_detector.yml

```diff
@@ -129,10 +125,63 @@ single-machine-performance-regression_detector:
     # uploading JUnit XML, so the upload command below respects that convention.
     - DATADOG_API_KEY="$("$CI_PROJECT_DIR"/tools/ci/fetch_secret.sh "$AGENT_API_KEY_ORG2" token)" || exit $?; export DATADOG_API_KEY
     - datadog-ci junit upload --service datadog-agent outputs/junit.xml
-    # Finally, exit 1 if the job signals a regression else 0.
-    - RUST_LOG="${RUST_LOG}" ./smp --team-id ${SMP_AGENT_TEAM_ID} --api-base ${SMP_API} --aws-named-profile ${AWS_NAMED_PROFILE}
-      job result
-      --submission-metadata submission_metadata
+    # Run quality gate check script
+    - |
+      python3 <<'EOF'
+      import json
+      import sys
+
+      try:
+          with open('outputs/report.json') as f:
+              data = json.load(f)
+      except FileNotFoundError:
+          print("Machine readable report not found.")
+          sys.exit(1)
+      except json.JSONDecodeError as e:
+          print(f"Error parsing JSON report: {e}")
+          sys.exit(1)
+
+      experiments = data.get('experiments', {})
+      failed = False
+      decision_record = []
+
+      for exp_name, exp_data in experiments.items():
+          if exp_name.startswith('quality_gate_'):
+              bounds_checks = exp_data.get('bounds_checks', {})
+              for check_name, check_data in bounds_checks.items():
+                  results = check_data.get('results', {})
+                  comparison = results.get('comparison', [])
+                  failed_replicates = [
+                      replicate for replicate in comparison if not replicate.get('passed', False)
+                  ]
+                  if failed_replicates:
+                      replicate_indices = [str(replicate.get('replicate_idx')) for replicate in failed_replicates]
+                      decision_record.append(
+                          f"- **{exp_name}**, bounds check **{check_name}**: Replicates {', '.join(replicate_indices)} **FAILED**."
+                      )
+                      failed = True
+                  else:
+                      decision_record.append(
+                          f"- **{exp_name}**, bounds check **{check_name}**: All replicates passed."
+                      )
+
+      with open('outputs/decision_record.md', 'w') as f:
+          # Extra newline since this is appended to another report
+          f.write('\n\n## CI Pass/Fail Decision\n\n')
```

Comment:
I think there's a case for putting this at the top of the report because it's arguably the most important piece of information for users.

Reply:
hmm, I could be convinced, don't feel strongly about it myself

Reply:
I'm imagining the user experience when someone's PR fails; I think putting this at the top helps emphasize its importance and ensure it's the first thing people see. That's especially helpful for guiding investigations when multiple things fail.

I think it could fit nicely under a `# Quality Gates Results` header.

Along the same lines, it'd be helpful to have bounds check dashboard links here or a note that links can be found in the bounds checks section under the respective experiment names.